lib/mix/lib/mix/tasks/release.ex

```diff
@@ -0,0 +1,916 @@
+  so you can erase all source after the release is assembled:
+
+      git clone remote://path/to/my_app.git my_app_source
+      cd my_app_source
+      mix deps.get --only prod
+      MIX_ENV=prod mix release --path ../my_app_release
+      cd ../my_app_release
+      rm -rf ../my_app_source
+      bin/start
+
+  However, this option can be expensive if you have multiple nodes or
+  if the release assembling process is a long one, as each node needs
+  to individually assemble the release.
+
+  You can automate this process in a couple different ways. One option
+  is to make it part of your Continuous Integration (CI) / Continuous
+  Deployment (CD) pipeline. When you have a CI/CD pipeline, it is common
+  that the machines in your CI/CD pipeline run on the exact same operating
+  system and tooling than your production servers (if they don't, they should).
+  In this case, you can assemble the release at the end of your CI/CD
+  pipeline and push it to S3 or any other network storage. To perform the
+  deployment, your production machines can fetch the deployment from the
+  network storage.
+
+  Another mechanism to automate deployments is to use images, such as
+  Amazon Machine Images, or container platforms, such as Docker.
+  For instance, you can use Docker to locally run the same operating
+  system that you have in your production servers, allowing you to
+  assemble a release or build a container or image locally, which is
+  then pushed to production. One can also build images and/or containers
+  at the end of their CI/CD pipelines.
+
+  In other words, there are multiple ways systems can be deployed and
+  releases can be automated and incorporated into all of them as long
+  as you remember that the release must be assembled in the same
+  operating system distribution and version as the production server.
+
+  Once a system is deployed, shutting down said systems can be done by
+  sending SIGINT/SIGTERM to the system, which is what most containers,
+  platforms and tools do, or by explicitly invoking `bin/RELEASE_NAME stop`.
+  Shutting down the system consists of stopping each application and
+  their respective supervisiont trees, one by one, in the opposite order
+  that they were started.
+
+  ## Options
+
+  Inside each release, here are the built-in options:
+
+    * `:applications` - a keyword list that configures and adds new applications
+      to the release. The key is the application name and the value is one of:
+
+        * `:permanent` - the application is started and the node shuts down
+          if the application terminates, regardless of reason
+        * `:transient` - the application is started and the node shuts down
+          if the application terminates abnormally
+        * `:temporary` - the application is started and the node does not
+          shut down if the application terminates
+        * `:load` - the application is only loaded
```

Comment:
One thing that bears mentioning here is that if an application is set to `:load` or `:none`, but is a dependency of other applications in the release, then those dependents will not start, even if declared `:permanent`; instead the release will just get stuck halfway through init, waiting for an application to start that never will.

The result of the above is that a release will appear to have started normally, but upon inspection will seem to be missing functionality (for example, perhaps a Phoenix endpoint isn't available, because the application containing isn't started yet due to waiting on some other application).

In my opinion, neither of these options should be made available without some static analysis which guarantees that the applications marked load/none will not produce a broken release.

Reply:
Good call. I will have added a TODO for the static check.